pkg/build/controller/build/build_controller.go

```diff
@@ -1581,8 +1583,12 @@ func (bc *BuildController) handleControllerConfig() error {
 	}
 	if config == nil {
 		bc.additionalTrustedCAData = nil
+		bc.defaultProxyConfig = v1.ProxyConfig{}
```

Comment:
you won't need to do this after the API bump - `ProxySpec` has normal `string` members, not string pointers.